allennlp/data/tokenizers/word_filter.py

```diff
@@ -33,38 +33,71 @@ def filter_words(self, words: List[Token]) -> List[Token]:
         return words
 
 
+@WordFilter.register('regex')
+class RegexFilter(WordFilter):
+    """
+    Uses a list of regexes to filter
```

Comment:
This needs a better docstring, including listing the parameters with comments describing them.  In particular, do these patterns say what to keep, or what to remove?